toonz/sources/toonz/ruler.cpp

```diff
@@ -274,6 +274,12 @@ void Ruler::mousePressEvent(QMouseEvent *e) {
       // muove la guida vecchia
       if (selected < count - 1) std::swap(guides[selected], guides.back());
       // aggiorna sprop!!!!
+      if (e->button() == Qt::RightButton) {
+        assert(!getGuides().empty());
+        getGuides().pop_back();
+        update();
+        m_viewer->update();
+      }
```

Comment:
@yehjaemin Thank you for adding a feature and so sorry for my late review.

For now you can remove the guide when you right-click anywhere in the ruler, regardless of the distance between the clicked position and the nearest guide to be deleted.

It seems to be useful to remove all guides, since you just need right-clicking multiple times without moving the cursor. But it may have a risk to accidentally remove the guide placed at the outside of the visible area.

Is it an intended behavior? Or should it have a condition that the deletion will be executed only when the clicked position is near the guide ( like `minDist2 <= 25` ) ?